src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettings.java

```diff
@@ -148,6 +153,19 @@ public void setRulesByKey(Map<String, Rule> rules) {
     this.rulesByKey = new HashMap<>(rules);
   }
 
+  public boolean isFocusOnNewCode() {
+    return isFocusOnNewCode;
+  }
+
+  public void setFocusOnNewCode(String place, DataContext dataContext, boolean focusOnNewCode) {
+    isFocusOnNewCode = focusOnNewCode;
+    var action = (SonarFocusOnNewCode) ActionManager.getInstance().getAction("SonarLint.SetFocusNewCode");
+    if (action.isFocusOnNewCode() != isFocusOnNewCode) {
+      var anActionEvent = AnActionEvent.createFromDataContext(place, null, dataContext);
+      action.setSelected(anActionEvent, focusOnNewCode);
+    }
```

Comment:
I'm not sure in terms of responsibility if this class is the right place to hold this, we usually have mostly setters and getters here